### PR TITLE
MGMT-15204: Add Unit tests infrastructure + first pkg/registry tests

### PR DESCRIPTION
**Pull request by nmagnezi**

### CODE REVIEW

**Comment by nmagnezi** [comment]
/hold
adding more tests

File: Makefile [hunk 1]
```diff
@@ -60,3 +73,27 @@ $(REPORTS):
 
 clean:
 	-rm -rf $(REPORTS)
+
+generate-mocks:
+	find . -name 'mock_*.go' -type f -not -path './vendor/*' -delete
+	go generate -v $(shell go list ./...)
+
+unit-test:
+	SKIP_UT_DB=1 $(MAKE) _unit_test TIMEOUT=30m TEST="$(or $(TEST),$(shell go list ./...))"
```
Comment (jhernand) [inline Makefile:82]: I think this `SKIP_UT_DB` variable isn't needed.

File: pkg/registry/registry.go [hunk 2]
```diff
@@ -26,30 +25,43 @@ type Registry interface {
 	StopRegistry() error
 }
 
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
```
Comment (jhernand) [inline pkg/registry/registry.go:30]: This looks very much like [http.RoundTripper](https://pkg.go.dev/net/http#RoundTripper).

File: pkg/registry/registry_test.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+				Executer:   mockExecuter,
+				HTTPClient: testClient,
+			})
+
+		err = imageRegistry.StartRegistry("")
+		Expect(err).ToNot(HaveOccurred())
+	})
+})
+
+func TestValidator(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (nmagnezi) [reply pkg/registry/registry_test.go]: Done

File: pkg/registry/registry.go (hunk 2 above)
Comment (nmagnezi) [reply pkg/registry/registry.go:30]: How is it better?

File: pkg/registry/registry_test.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
```
[the remark this answers is not in the document]
Comment (nmagnezi) [reply pkg/registry/registry_test.go]: Done

File: go.mod [hunk 5]
```diff
@@ -140,21 +167,26 @@ require (
 	github.com/mgutz/ansi v0.0.0-20170206155736-9520e82c474b // indirect
 	github.com/mitchellh/go-homedir v1.1.0 // indirect
 	github.com/mitchellh/mapstructure v1.5.0 // indirect
+	github.com/moby/term v0.0.0-20220808134915-39b0c02b01ae // indirect
 	github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822 // indirect
 	github.com/nutanix-cloud-native/prism-go-client v0.2.1-0.20220804130801-c8a253627c64 // indirect
 	github.com/oklog/ulid v1.3.1 // indirect
+	github.com/onsi/ginkgo v1.16.5
```
[the remark this answers is not in the document]
Comment (nmagnezi) [reply go.mod]: Done

**Pull request merged**